UnitySDK/Assets/ML-Agents/Scripts/Brain.cs

```diff
@@ -21,27 +22,19 @@ public abstract class Brain : ScriptableObject
         protected Dictionary<Agent, AgentInfo> m_AgentInfos =
             new Dictionary<Agent, AgentInfo>(1024);
 
-        protected Batcher m_BrainBatcher;
+        protected ICommunicator m_Communicator;
 
         [System.NonSerialized]
         private bool m_IsInitialized;
 
         /// <summary>
-        /// Sets the Batcher of the Brain. The brain will call the batcher at every step and give
-        /// it the agent's data using SendBrainInfo at each DecideAction call.
+        /// Sets the Batcher of the Brain. The brain will call the communicator at every step and give
+        /// it the agent's data using PutObservations at each DecideAction call.
         /// </summary>
-        /// <param name="batcher"> The Batcher the brain will use for the current session</param>
-        public void SetBatcher(Batcher batcher)
+        /// <param name="communicator"> The Batcher the brain will use for the current session</param>
+        public void SetCommunicator(ICommunicator communicator)
         {
-            if (batcher == null)
-            {
-                m_BrainBatcher = null;
-            }
-            else
-            {
-                m_BrainBatcher = batcher;
-                m_BrainBatcher.SubscribeBrain(name);
-            }
+            m_Communicator = communicator ?? null;
```

Comment:
c# noob question - how is this different from `m_Communicator = communicator;`? m_Communicator will end up null if communicator is null either way, right?

Reply:
yeah, this is gone now.